src/main/java/com/example/demo/controllers/LocationController.java

```diff
@@ -0,0 +1,55 @@
+package com.example.demo.controllers;
+
+import com.example.demo.entities.LocationEntity;
+import com.example.demo.model.dto.LocationCreationDTO;
+import com.example.demo.services.LocationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class LocationController {
+
+    @Autowired
+    LocationService locationService;
+
+    @PostMapping(path = "/location")
+    public LocationEntity createLocation(@RequestBody LocationCreationDTO locationDTO)
+    {
+        //include validation possibly.
+        LocationEntity location = locationDTO.convertDtoToEntity();
+        return locationService.createLocation(location);
+    }
+
+    @ResponseBody
+    @GetMapping(path = "/location/{id}")
+    public LocationEntity findById(@PathVariable Long id)
+    {
+    return locationService.findLocationById(id);
+    }
+
+    @ResponseBody
+    @GetMapping(path = "/location")
+    public List<LocationEntity> allLocationsOrdered(@RequestParam (value = "page", defaultValue = "0") int page,
```

Comment:
Does the Pageable page parameter here not get you all of this?

Reply:
I couldn't find a default constructor for Pageable:
"Cannot construct instance of `org.springframework.data.domain.Pageable` (no Creators, like default construct, exist)"

Reply:
I am sure there is a better way. I'll do some digging :-)